### PR TITLE
[otbn,rtl,dv] Add status for internal secure wipe

### PR DESCRIPTION
**Pull request by andreaskurth**
After execution OTBN securely wipes its internal state.  Prior to this
commit, OTBN remained in the `BUSY_EXECUTE` status during this secure
wipe.  The `BUSY_EXECUTE` status, however, is defined as "OTBN is busy
executing software", which is not accurate: when performing a secure
wipe, OTBN no longer executes software.  Additionally, there is work in
progress to perform an internal secure wipe after reset as well as on a
software command (the latter is already possible for the instruction and
the data memories).  The newly added status will be used to represent
those operations as well.

### CODE REVIEW

File: hw/ip/otbn/dv/uvm/env/otbn_env_pkg.sv
Comment (engdoreis) [inline hw/ip/otbn/dv/uvm/env/otbn_env_pkg.sv:121]: Nit. What if you move the `otbn_pkg::StatusBusySecWipeInt` after `otbn_pkg::StatusBusyExecute,` so that we would just see one line changing in the diff.
Feel free to ignore and mark it as resolved.

**Review by engdoreis — APPROVED** [review]
Thanks @andreaskurth for the PR.
The software part is ok, but I think you should execute the otbn chip level tests for precaution.

And wait for the approval of a design engineer before merging.

**Comment by andreaskurth** [comment]
> Thanks @andreaskurth for the PR. The software part is ok, but I think you should execute the otbn chip level tests for precaution.

Thanks for your review, @engdoreis! I have now executed the OTBN chip-level tests. They all pass in DV and three tests also pass in Verilator, but `clkmgr_off_otbn_trans_test_sim_verilator`, `otbn_ecdsa_op_irq_test_sim_verilator`, `otbn_irq_test_sim_verilator`, and `otbn_mem_scramble_test_sim_verilator` time out (at least on my laptop). Should this be investigated?

File: hw/ip/otbn/dv/uvm/env/otbn_env_pkg.sv
Comment (andreaskurth) [reply hw/ip/otbn/dv/uvm/env/otbn_env_pkg.sv:121]: The enum variants are defined in this order, and I think keeping this symmetry in the code is more important than minimizing the diff.

**Review by GregAC — APPROVED** [review]
LGTM @ctopal could you take a look at the DV changes?

**Comment by GregAC** [comment]
> hey all pass in DV and three tests also pass in Verilator, but clkmgr_off_otbn_trans_test_sim_verilator, otbn_ecdsa_op_irq_test_sim_verilator, otbn_irq_test_sim_verilator, and otbn_mem_scramble_test_sim_verilator time out (at least on my laptop). Should this be investigated?

They're not run in CI any more so might just be broken. Worth checking if they're similarly broken on master first. If it is this PR breaking them we should fix them first.

**Review by ctopal — APPROVED** [review]
LGTM, Modelling of it was already done before so this is indeed a low impact one for DV side. Thanks @andreaskurth!

**Comment by andreaskurth** [comment]
> > hey all pass in DV and three tests also pass in Verilator, but clkmgr_off_otbn_trans_test_sim_verilator, otbn_ecdsa_op_irq_test_sim_verilator, otbn_irq_test_sim_verilator, and otbn_mem_scramble_test_sim_verilator time out (at least on my laptop). Should this be investigated?
> 
> They're not run in CI any more so might just be broken. Worth checking if they're similarly broken on master first. If it is this PR breaking them we should fix them first.

Those tests also timeout on `master` (again on my laptop), so I am not investigating this further in this PR.

**Pull request merged**
